lib/rules/no-unused-vars.js

```diff
@@ -18,26 +18,49 @@ module.exports = {
       category: 'essential',
       url: 'https://eslint.vuejs.org/rules/no-unused-vars.html'
     },
-    fixable: null,
-    schema: []
+    fixable: 'code',
+    schema: [
+      {
+        'type': 'object',
+        'properties': {
+          'varIgnorePattern': {
+            'type': 'string'
+          }
+        },
+        'additionalProperties': false
+      }
+    ]
   },
 
   create (context) {
+    const option = context.options[0] || { varIgnorePattern: '^_' }
+    const pattern = option['varIgnorePattern'] || '^_'
+    // only use for construct a regularExpression
+    // eslint-disable-next-line no-eval
+    const regExp = eval(`/${pattern}/`)
     return utils.defineTemplateBodyVisitor(context, {
       VElement (node) {
         const variables = node.variables
 
         for (
           let i = variables.length - 1;
-          i >= 0 && !variables[i].references.length;
+          i >= 0 && !regExp.test(variables[i].id.name) && !variables[i].references.length;
           i--
         ) {
           const variable = variables[i]
           context.report({
             node: variable.id,
             loc: variable.id.loc,
             message: `'{{name}}' is defined but never used.`,
-            data: variable.id
+            data: variable.id,
+            suggest: [
+              {
+                desc: 'Replace the unused-var with default ignore pattern',
+                fix: function (fixer) {
+                  return fixer.replaceText(variable.id, `_${variable.id.name}`)
```

Comment:
I think it is good to use SuggestionAPI.

However, this suggestion is only valid if `'^_'` is set.